hicommonwealth/commonwealth
# Integrate user mentions with Knock

## Link to Issue
Closes: #7687

## Description of Changes

## Test Plan
- Add `UserMentioned` to `ALLOWED_EVENTS` env var in your `commonwealth/.env` file
- Set the `KNOCK_API_KEY` in your `.env` file (DM me to get the key)
- Run `yarn start-knock` in `packages/commonwealth/`
- In another terminal run `yarn start-message-relayer` in `packages/commonwealth`
- Start the Commonwealth app in another terminal
- Sign in on one account and create a thread
  - note the user id of this account (`SELECT user_id FROM "Addresses" WHERE address = '[your address]';`)
- On another account, comment on the thread you created and mention the author of the thread
- Now, join the common-dev Knock workspace (just create an account with your `@common.xyz` email)
- Go to https://dashboard.knock.app/common-dev/development/developers/logs/runs and look for a workflow with the recipient as the user id noted above and the workflow as 'user-mentioned'
- Once you find the workflow run, click on it and then click on the `In-App` step and then click on the hyperlinked 'message' link. Once there, switch to the 'Preview' tab and make sure the comment appears as it should

## Deployment Plan


## Other Considerations

-
